packages/components/src/slot-fill/types.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * External dependencies
+ */
+import type { Component, MutableRefObject, ReactNode, RefObject } from 'react';
+
+export type SlotKey = string | symbol;
+
+export type FillProps = any;
+
+type SlotPropBase = {
+	/**
+	 * Slot name.
+	 */
+	name: SlotKey;
+	/**
+	 * props to pass from `Slot` to `Fill`.
+	 *
+	 * @default {}
+	 */
+	fillProps?: FillProps;
+
+	/**
+	 * By default, events will bubble to their parents on the DOM hierarchy (native event bubbling).
+	 * If set to true, events will bubble to their virtual parent in the React elements hierarchy instead,
+	 * also accept an optional `className`, `id`, etc.  to add to the slot container.
+	 */
+	bubblesVirtually?: boolean;
+};
+
+export type SlotComponentProps =
+	| ( SlotPropBase & {
+			bubblesVirtually: true;
```

Comment:
For extra clarity?

```suggestion
			bubblesVirtually: true;
			children?: never;
```

Reply:
If added `children?: never` , I get an error on the following line.

https://github.com/WordPress/gutenberg/blob/48890827dfd74dd536c2ce24e0a0a5f7a69882d4/packages/components/src/slot-fill/index.tsx#L78

Is there any way to make it better?

Reply:
Yup, that's because the `Omit` utility doesn't do well with type unions.

Something like this should fix it:

```diff
diff --git a/packages/components/src/slot-fill/index.tsx b/packages/components/src/slot-fill/index.tsx
index 57a9bfbe58..043b87c1b9 100644
--- a/packages/components/src/slot-fill/index.tsx
+++ b/packages/components/src/slot-fill/index.tsx
@@ -67,6 +67,14 @@ export function Provider( { children }: SlotFillProviderProps ) {
 	);
 }
 
+type DistributiveOmit< T, K extends keyof any > = T extends any
+	? Omit< T, K >
+	: never;
+type SlotComponentPropsWithoutName = DistributiveOmit<
+	SlotComponentProps,
+	'name'
+>;
+
 export function createSlotFill( key: SlotKey ) {
 	const baseName = typeof key === 'symbol' ? key.description : key;
 	const FillComponent = ( props: Omit< FillComponentProps, 'name' > ) => (
@@ -74,7 +82,7 @@ export function createSlotFill( key: SlotKey ) {
 	);
 	FillComponent.displayName = `${ baseName }Fill`;
 
-	const SlotComponent = ( props: Omit< SlotComponentProps, 'name' > ) => (
+	const SlotComponent = ( props: SlotComponentPropsWithoutName ) => (
 		<Slot name={ key } { ...props } />
 	);
 	SlotComponent.displayName = `${ baseName }Slot`;
diff --git a/packages/components/src/slot-fill/types.ts b/packages/components/src/slot-fill/types.ts
index c16715d6fe..da07dac01d 100644
--- a/packages/components/src/slot-fill/types.ts
+++ b/packages/components/src/slot-fill/types.ts
@@ -28,6 +28,7 @@ export type SlotComponentProps =
 			 * also accept an optional `className`, `id`, etc.  to add to the slot container.
 			 */
 			bubblesVirtually: true;
+			children?: never;
 	  } )
 	| ( SlotPropBase & {
 			/**

```

Source: https://stackoverflow.com/questions/57103834/typescript-omit-a-property-from-all-interfaces-in-a-union-but-keep-the-union-s

Reply:
@ciampo 
Sorry for the delay in getting back to you.

Excellent! Works correctly!!